l1-contracts/deploy-scripts/upgrade/DefaultEcosystemUpgrade.s.sol

```diff
@@ -237,6 +260,44 @@ contract DefaultEcosystemUpgrade is Script, DeployL1Script {
         notifyAboutDeployment(contractAddress, contractName, creationCalldata, contractName, true);
     }
 
+    function deployGWTuppWithContract(string memory contractName) internal returns (address proxyAddress) {
+        bytes memory creationCalldata = getCreationCalldata(contractName, true);
+        address implementation = Utils.deployThroughL1Deterministic(
+            getCreationCode(contractName, true),
+            creationCalldata,
+            0,
+            newConfig.priorityTxsL2GasLimit,
+            new bytes[](0),
+            gatewayConfig.chainId,
+            addresses.bridgehub.bridgehubProxy,
+            addresses.bridges.l1AssetRouterProxy
+        );
+        notifyAboutDeployment(implementation, contractName, creationCalldata, contractName, true);
```

Comment:
what does notifyAboutDeployment do here? we should add a GW flag, so that we can provide GW rpc url later